core-web/libs/template-builder/src/lib/components/template-builder/components/remove-confirm-dialog/remove-confirm-dialog.component.ts

```diff
@@ -18,13 +24,25 @@ import { DotMessagePipe, DotMessagePipeModule } from '@dotcms/ui';
 export class RemoveConfirmDialogComponent {
     @Output() deleteConfirmed: EventEmitter<void> = new EventEmitter();
     @Output() deleteRejected: EventEmitter<void> = new EventEmitter();
+    private currentPopup: ConfirmationService;
+
+    @HostListener('document:keydown.escape', ['$event'])
+    onEscapePress() {
+        if (this.currentPopup) {
+            this.currentPopup.close();
+            this.deleteRejected.emit();
+            this.currentPopup = null;
+        }
+    }
+
```

Comment:
why do we need this?

Reply:
Because the confirmPopup don't handle this by itself.